aptos-move/aptos-vm-logging/src/lib.rs

```diff
@@ -65,30 +83,59 @@ pub fn init_speculative_logs(num_txns: usize) {
 /// events storage is not initialized or appropriately sized.
 pub fn speculative_log(level: Level, context: &AdapterLogSchema, message: String) {
     let txn_idx = context.get_txn_idx();
```

Comment:
we can just check for `speculation_disabled` on the top of this function?

Reply:
It's an extra atomic read that I wanted to save :)

Reply:
lmk if that's okay

Reply:
It just makes the code a bit unclean, but may be worth saving the extra atomic read :)

Reply:
changed this one. In the other ones, we don't have context, so there seems to be no code duplication.